source/agora/consensus/data/Block.d

```diff
@@ -574,8 +569,29 @@ version (unittest)
         // however many tests calling makeNewTestBlock have no access to ConsensusParams
         auto block = makeNewBlock(prev_block, txs,
                 time_offset ? time_offset : prev_block.header.time_offset + 1,
-                random_seed, enrollments, missing_validators);
-        return multiSigTestBlock(block, cycleForValidator, key_pairs);
+                random_seed, key_pairs.length, enrollments, missing_validators);
+        auto validators = BitField!ubyte(key_pairs.length);
+        Signature[] sigs;
+        ulong offset = 0;
+        key_pairs.enumerate.each!((i, k)
+        {
+            validators[i] = true;
+            sigs ~= block.createBlockSignature(k.secret, offset);
+        });
+        try
+        {
+            auto signed_block = block.updateSignature(multiSigCombine(sigs), validators);
+            return signed_block;
+        }
+        catch (Exception e)
+        {
+            () @trusted
+            {
+                import std.format;
+                assert(0, format!"makeNewTestBlock exception thrown during test: %s"(e));
+            }();
+        }
```

Comment:
Why does this function need to be `nothrow` ?

Reply:
The code in Nominator that calls it is `nothrow`

Reply:
No actually not called from `Nominator` but called from test code which was `nothrow`

Reply:
Why would `Nominator` call `makeNewTestBlock` ? :O

Reply:
I thought it best to keep the test code checking that the code is `nothrow`.